src/test/java/de/tum/in/www1/artemis/text/TextAssessmentIntegrationTest.java

```diff
@@ -319,60 +300,52 @@ private TextSubmission prepareSubmission() throws Exception {
         return request.get("/api/exercises/" + textExercise.getId() + "/text-submission-without-assessment", HttpStatus.OK, TextSubmission.class, params);
     }
 
-    @Test
+    @ParameterizedTest(name = "{displayName} [{index}] {argumentsWithNames}")
     @WithMockUser(username = TEST_PREFIX + "tutor1", roles = "TA")
-    void saveTextAssessment_studentHidden() throws Exception {
+    @ValueSource(booleans = { true, false })
+    void saveOrSubmitTextAssessment_studentHidden(boolean submit) throws Exception {
         TextSubmission submissionWithoutAssessment = prepareSubmission();
-
         final TextAssessmentDTO textAssessmentDTO = new TextAssessmentDTO();
         textAssessmentDTO.setFeedbacks(new ArrayList<>());
 
-        Result result = request.putWithResponseBody("/api/participations/" + submissionWithoutAssessment.getParticipation().getId() + "/results/"
-                + submissionWithoutAssessment.getLatestResult().getId() + "/text-assessment", textAssessmentDTO, Result.class, HttpStatus.OK);
+        Result result;
+        if (submit) {
+            result = request.postWithResponseBody(
+                    "/api/participations/" + submissionWithoutAssessment.getParticipation().getId() + "/results/"
+                            + Objects.requireNonNull(submissionWithoutAssessment.getLatestResult()).getId() + "/submit-text-assessment",
+                    textAssessmentDTO, Result.class, HttpStatus.OK);
+
+        }
+        else {
+            result = request.putWithResponseBody("/api/participations/" + submissionWithoutAssessment.getParticipation().getId() + "/results/"
+                    + Objects.requireNonNull(submissionWithoutAssessment.getLatestResult()).getId() + "/text-assessment", textAssessmentDTO, Result.class, HttpStatus.OK);
+        }
```

Comment:
This could probably also be a method since it is used here and in the test below.

Reply:
good call!